app/src/main/java/org/eyeseetea/malariacare/network/ServerAPIController.java

```diff
@@ -719,15 +722,29 @@ public static OrganisationUnit getOrgUnitByPhone(String imei) throws ApiCallExce
             //{"organisationUnits":[{}]}
             JSONArray orgUnitsArray = (JSONArray) body.get(TAG_ORGANISATIONUNITS);
 
+            boolean moreThanOneOrgUnit = false;
+
             //0| >1 matches -> Error
-            if (orgUnitsArray.length() == 0 || orgUnitsArray.length() > 1) {
+            if (orgUnitsArray.length() == 0) {
                 Log.e(TAG, String.format("getOrgUnitData(%s) -> Found %d matches", imei,
                         orgUnitsArray.length()));
                 return null;
+            } else if (orgUnitsArray.length() > 1) {
+                moreThanOneOrgUnit = true;
```

Comment:
@xurxodev do we need this flag? couldn't we just ask for orgUnitsArray.length()>1 down in the condition to throw the exception?